ts/nni_manager/training_service/reusable/environments/dlcEnvironmentService.ts

```diff
@@ -0,0 +1,146 @@
+    }
+
+    public createEnvironmentInformation(envId: string, envName: string): EnvironmentInformation {
+        return new DlcEnvironmentInformation(envId, envName);
+    }
+
+    public get getName(): string {
+        return 'dlc';
+    }
+
+    public async refreshEnvironmentsStatus(environments: EnvironmentInformation[]): Promise<void> {
+        environments.forEach(async (environment) => {
+            const dlcClient = (environment as DlcEnvironmentInformation).dlcClient;
+            if (!dlcClient) {
+                return Promise.reject('DLC client not initialized!');
+            }
+            const newStatus = await dlcClient.updateStatus(environment.status);
+            switch (newStatus.toUpperCase()) {
+                case 'CREATING':
+                case 'CREATED':
+                case 'WAITING':
+                case 'QUEUED':
+                    environment.setStatus('WAITING');
+                    break;
+                case 'RUNNING':
+                    environment.setStatus('RUNNING');
+                    break;
+                case 'COMPLETED':
+                case 'SUCCEEDED':
+                    environment.setStatus('SUCCEEDED');
+                    break;
+                case 'FAILED':
+                    environment.setStatus('FAILED');
+                    return Promise.reject(`DLC: job ${environment.envId} is failed!`);
+                case 'STOPPED':
+                case 'STOPPING':
+                    environment.setStatus('USER_CANCELED');
+                    break;
+                default:
+                    environment.setStatus('UNKNOWN');
+            }
+        });
+    }
+
+    public async startEnvironment(environment: EnvironmentInformation): Promise<void> {
+        const dlcEnvironment: DlcEnvironmentInformation = environment as DlcEnvironmentInformation;
+        const environmentLocalTempFolder = path.join(this.experimentRootDir, "environment-temp");
+        if (!fs.existsSync(environmentLocalTempFolder)) {
+            await fs.promises.mkdir(environmentLocalTempFolder, {recursive: true});
+        }
+
+        const dlcFolder: string = this.experimentRootDir.replace(
+            this.config.localStorageMountPoint, this.config.containerStorageMountPoint);
+        dlcEnvironment.workingFolder = `${this.experimentRootDir}/envs/${environment.id}`;
+        dlcEnvironment.runnerWorkingFolder = `${dlcFolder}/envs/${environment.id}`;
+        let script: string = environment.command;
+
+        // environment id dir and command dir
+        if (!fs.existsSync(`${dlcEnvironment.workingFolder}/commands`)) {
+            await fs.promises.mkdir(`${dlcEnvironment.workingFolder}/commands`, {recursive: true});
+        }
+
+        const prepare = `cd ${dlcEnvironment.runnerWorkingFolder} && cp -r ../../environment-temp/envs/* ../`;
+        const startrun = `sh ../install_nni.sh && python -m nni.tools.trial_tool.trial_runner`;
```

Comment:
python -> python3?

Reply:
in default DLC container image, python is python3.